src/mpt/parseMPT.ts

```diff
@@ -53,17 +64,57 @@ function parseData(data: string[]): MPT['variables'] {
   return variables;
 }
 
-interface HeaderMPT {
-  fileType: string;
-  Technique: string;
-  [other: string]: string;
-}
+/**
+ * Creates an mps object from an mps file
+ * The output is similar, but not the same, than `MPR.settings`
+ * MPS includes one or more techniques
+ * We will know what this means when using it.
+ *
+ * @param data - pass the file as string, Buffer or Arraybuffer.
+ * @returns JSON object representing the parsed data
+ */
 
-function headerMPT(lines: string[]): HeaderMPT {
-  const nOfLinesHeader = lines[1].split(' : ');
-  return {
-    fileType: lines[0],
-    'Nb header lines': nOfLinesHeader[1].trim() || '',
-    Technique: lines[3],
+export function parseMPTSettings(
+  lines: string[],
+  technique: string,
+): ComplexObject {
+  // file converted to an array of strings, each item a newline.
+
+  let result: ComplexObject = { technique, params: [] };
+
+  const regex = {
+    nothing: /^\s*$/,
+    keyValue: / : | :$/,
+    multiline: /^[ \t]/,
+    table: /^\w.*\s{2,}-*\w+.*\s{4,}$/,
   };
+
+  for (let i = 0; i < lines.length; i++) {
+    const currentLine = lines[i];
+
+    if (regex.nothing.test(currentLine)) {
+      continue;
+    } else if (regex.keyValue.test(currentLine)) {
+      //function updates the index bc some values are multiline
+      [result, i] = addKeyValueToResult(
+        result,
+        lines,
+        currentLine,
+        regex,
+        i,
+        'MPT',
+      );
+    } else if (regex.table.test(currentLine)) {
+      /* for not k : v */
+      //regex.table
+      const kV: string[] = currentLine.split(/\s{2,}/);
+      const [key, val] = [kV[0].trim(), kV.slice(1).join('  ').trim()];
```

Comment:
Is it normal you want to join with 2 spaces ?

Reply:
Thanks for all the fixes. In this case yes, I will comment back tomorrow. Adding a few more commits now.